libraries/shared/storage/fetcher/geth_rpc_storage_fetcher.go

```diff
@@ -66,45 +61,177 @@ func (fetcher GethRpcStorageFetcher) FetchStorageDiffs(out chan<- types.RawDiff,
+
+		for _, accountStorage := range account.Storage {
+			rawDiff, formatErr := types.FromOldGethStateDiff(account, stateDiff, accountStorage)
+			if formatErr != nil {
+				errs <- formatErr
 			}
+
+			logrus.Tracef("adding storage diff to out channel. keccak of address: %v, block height: %v, storage key: %v, storage value: %v",
+				rawDiff.HashedAddress.Hex(), rawDiff.BlockHeight, rawDiff.StorageKey.Hex(), rawDiff.StorageValue.Hex())
+			out <- rawDiff
 		}
+	}
+}
 
+func (fetcher GethRpcStorageFetcher) handleDiffsFromNewGethPatchWithService(payload filters.Payload, out chan<- types.RawDiff, errs chan<- error) {
+	stateDiff, decodeErr := fetcher.decodeStateDiffRlpFromPayloadForOldPatches(payload)
+	if decodeErr != nil {
+		errs <- decodeErr
+	}
+
+	for _, account := range stateDiff.UpdatedAccounts {
+		logrus.Trace(fmt.Sprintf("iterating through %d Storage values on account", len(account.Storage)))
+
+		for _, accountStorage := range account.Storage {
+			rawDiff, formatErr := types.FromNewGethStateDiff(account, stateDiff, accountStorage)
+			if formatErr != nil {
+				errs <- formatErr
+			}
+
+			logrus.Tracef("adding storage diff to out channel. keccak of address: %v, block height: %v, storage key: %v, storage value: %v",
+				rawDiff.HashedAddress.Hex(), rawDiff.BlockHeight, rawDiff.StorageKey.Hex(), rawDiff.StorageValue.Hex())
+			out <- rawDiff
+		}
 	}
 }
 
-func (fetcher GethRpcStorageFetcher) formatDiff(account statediff.AccountDiff, stateDiff *statediff.StateDiff, storage statediff.StorageDiff) (types.RawDiff, error) {
-	if fetcher.gethVersion == OldGethPatch {
-		return types.FromOldGethStateDiff(account, stateDiff, storage)
+func (fetcher GethRpcStorageFetcher) handleDiffsFromNewGethPatchWithFilter(payload filters.Payload, out chan<- types.RawDiff, errs chan<- error) {
+	var stateDiff filters.StateDiff
+	decodeErr := rlp.DecodeBytes(payload.StateDiffRlp, &stateDiff)
+	if decodeErr != nil {
+		errs <- decodeErr
+	}
+
+	for _, account := range stateDiff.UpdatedAccounts {
+		logrus.Trace(fmt.Sprintf("iterating through %d Storage values on account", len(account.Storage)))
+
+		for _, accountStorage := range account.Storage {
+			rawDiff, formatErr := types.FromNewGethStateDiff(account, &stateDiff, accountStorage)
+			if formatErr != nil {
+				errs <- formatErr
+			}
+
+			logrus.Tracef("adding storage diff to out channel. keccak of address: %v, block height: %v, storage key: %v, storage value: %v",
+				rawDiff.HashedAddress.Hex(), rawDiff.BlockHeight, rawDiff.StorageKey.Hex(), rawDiff.StorageValue.Hex())
+			out <- rawDiff
+		}
+	}
+}
+
+func (fetcher GethRpcStorageFetcher) decodeStateDiffRlpFromPayload(payload filters.Payload) (*filters.StateDiff, error) {
+	if fetcher.gethVersion == NewGethPatchWithFilter {
+		var stateDiff filters.StateDiff
+		decodeErr := rlp.DecodeBytes(payload.StateDiffRlp, &stateDiff)
+		if decodeErr != nil {
+			return &filters.StateDiff{}, decodeErr
```

Comment:
are we able to return `nil` instead of `&filters.StateDiff{}`? Or is this to circumvent a null pointer error above?